lib/improver/spotdata/tests/test_neighbour_finding.py

```diff
@@ -0,0 +1,742 @@
+        """Test plugins behaviour with no ancillary data provided"""
+        plugin = PointSelection(method, vertical_bias, land_constraint)
+        if method == 'fast_nearest_neighbour':
+            result = plugin.process(self.cube, self.sites, {})
+            self.assertIsInstance(result, np.ndarray)
+        else:
+            msg = 'Ancillary data'
+            with self.assertRaisesRegexp(Exception, msg):
+                result = plugin.process(self.cube, self.sites, {})
+
+
+class miscellaneous(TestNeighbourFinding):
+    def test_invalid_method(self):
+        """
+        Test that the plugin can handle an invalid method being passed in.
+
+        """
+        plugin = PointSelection('smallest distance')
+        msg = 'Unknown method'
+        with self.assertRaisesRegexp(AttributeError, msg):
+            plugin.process(self.cube, self.sites, self.ancillary_data)
+
+    def test_variable_no_neighbours(self):
+        """
+        Test that the plugin can handle a variable number of neigbours to use
+        when relaxing the 'nearest' condition. Make the smallest displacement
+        point 2-grid cells away, so it should be captured with no_neighbours
+        set to 25.
+
+        """
+        self.ancillary_data['orography'].data[13, 10] = 10.
+        plugin = PointSelection(method='minimum_height_error_neighbour',
+                                vertical_bias=None,
+                                land_constraint=False)
+        result = plugin.process(self.cube, self.sites, self.ancillary_data,
+                                no_neighbours=25)
+        self.assertEqual(result['i'], 13)
+        self.assertEqual(result['j'], 10)
+        self.assertEqual(result['dz'], 0.)
+
+    def test_invalid_no_neighbours(self):
+        plugin = PointSelection(method='minimum_height_error_neighbour',
+                                vertical_bias=None,
+                                land_constraint=False)
+        msg = 'Invalid nearest no'
+        with self.assertRaisesRegexp(ValueError, msg):
+            plugin.process(self.cube, self.sites, self.ancillary_data,
+                           no_neighbours=20)
+
+
+class fast_nearest_neighbour(TestNeighbourFinding):
+    '''
+    Tests for fast_nearest_neighbour method. No other conditions beyond
+    proximity are considered.
+
+    '''
+    method = 'fast_nearest_neighbour'
+
+    def test_return_type(self):
+        '''Ensure a numpy array of the format expected is returned.'''
+        self.return_types(self.method)
+
+    def test_correct_neighbour(self):
+        '''Nearest neighbouring grid point with no other conditions'''
+        self.correct_neighbour(self.method, 15, 10, 10.)
+
+    def test_without_ancillary_data(self):
+        '''
+        Should function without any ancillary fields and return expected type.
+        '''
+        self.without_ancillary_data(self.method)
+
+
+class min_dz_no_bias(TestNeighbourFinding):
```

Comment:
There are no longer methods called `min_dz_no_bias` within your `PointSelection` class. It might be best to move all these tests under a `Test_minimum_height_error_neighbour` class with different sections, to match the style of pre-existing tests.